public.go

```diff
@@ -64,3 +64,14 @@ func (p *pub) Wipe() {
 	p.pre = '0'
 	io.ReadFull(rand.Reader, p.pub)
 }
+
+func (p *pub) Seal(input []byte, recipient string) ([]byte, error) {
+	return nil, ErrCannotSeal
```

Comment:
We might have the private key but its wrong type. I think errors need to be better for these.

Reply:
in the case above, we only have a public key - this is the public key version of KeyPair. The message is `nkeys: cannot seal no private curve key available` which aligns with the equivalent one for Sign which is `nkeys: can not sign, no private key available`